src/osd/PGLog.h

```diff
@@ -289,13 +289,20 @@ struct PGLog : DoutPrefixProvider {
       }
       p = extra_caller_ops.find(r);
       if (p != extra_caller_ops.end()) {
+	uint32_t idx = 0;
 	for (auto i = p->second->extra_reqids.begin();
 	     i != p->second->extra_reqids.end();
-	     ++i) {
+	     ++idx, ++i) {
 	  if (i->first == r) {
 	    *version = p->second->version;
 	    *user_version = i->second;
 	    *return_code = p->second->return_code;
+	    if (*return_code >= 0) {
```

Comment:
maybe == 0 instead of >= 0?  I'm not sure we ever store a >0 value, but if we did, it seems like we might want to return it?

Reply:
@jdurgin or maybe we should prefer negative error codes here?

Reply:
I guess it's a moot point since this is always going to be a promote log entry, where we know the return_code is always 0

Reply:
> return_code is always 0

OK, I didn't know if that was true or not and I was trying to allow any cache tier PG log failures supersede the base tier return codes.